Sources/ContainerBuilder/ContainerBuilder+PullImage.swift

```diff
@@ -0,0 +1,167 @@
+import Compression
+import ContainerRegistry
+import Foundation
+import Logging
+import Shell
+
+extension ContainerImageSpec {
+    private static let logger = Logger(label: "apache-edge.container-builder")
+
+    /// Creates a container image specification with a base image from a registry and an executable to add.
+    ///
+    /// - Parameters:
+    ///   - baseImage: Reference to the base image to use (e.g., "debian:bookworm-slim").
+    ///   - executable: The URL of the executable to include in the image.
+    ///   - workingDir: The working directory for the container.
+    ///   - env: Environment variables to set in the container.
+    ///   - created: The date when the image was created.
+    ///   - architecture: The target CPU architecture (e.g., "arm64", "amd64").
+    ///   - username: Optional username for registry authentication.
+    ///   - password: Optional password for registry authentication.
+    /// - Returns: A new ContainerImageSpec with base image layers and the executable.
+    public static func withBaseImage(
+        baseImage: String,
+        executable: URL,
+        workingDir: String = "/",
+        env: [String] = ["PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"],
+        created: Date = Date(),
+        architecture: String = "arm64",
+        username: String? = nil,
+        password: String? = nil
+    ) async throws -> ContainerImageSpec {
+        let imageRef = try ImageReference(fromString: baseImage, defaultRegistry: "docker.io")
+        logger.info("Pulling base image: \(imageRef)")
```

Comment:
nit: I'd recommend doing static log message and then `, metadate: ["imageRef": imageRef]` to make the logging structured. not a blocker right now, but our futures selves will thank us